app/gui/src/project-view/components/visualizations/tableVizToolbar.ts

```diff
@@ -16,14 +16,15 @@ export type SortModel = {
 export interface SortFilterNodesButtonOptions {
   filterModel: ToValue<{
     [key: string]: {
-      values: any[]
+      values: string[]
       filterType: string
     }
   }>
   sortModel: ToValue<SortModel[]>
   isDisabled: ToValue<boolean>
   isFilterSortNodeEnabled: ToValue<boolean>
-  createNodes: (...options: NodeCreationOptions[]) => void
+  createNodes: (...options: NodeCreationOptions[]) => void,
+  colTypeMap: Map<string, string>
```

Comment:
IMO mutable non-reactive data in a reactive API is a code smell. It avoids some unneeded reactivity overhead (usually for an event handler)--but it's only a half-measure: It requires the caller to eagerly keep the data up-to-date.

Rather than push data to event handlers, give them a way to pull it as needed--something like:
`getColumnValueToEnso: (columnName) => ((columnValue: string) => Ast.Owned<Ast.Expression>)`

Then in `TableVisualization`, the `colTypeMap` (that will be used internally by the `getColumnValueToEnso` implementation) can be a `computed` so that it is only constructed when an event handler needs the data.